include/rcutils/env.h

```diff
@@ -0,0 +1,47 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCUTILS__ENV_H_
+#define RCUTILS__ENV_H_
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include <stdbool.h>
+
+#include "rcutils/macros.h"
+#include "rcutils/visibility_control.h"
+
+/// Set or un-set a process-scoped environment variable.
+/**
+ * This function modifies the environment variables for the current process.
+ *
+ * \param[in] env_name Name of the environment variable to modify.
+ * \param[in] env_value Value to set the environment variable to, or NULL to un-set.
+ * \return `True` if success
+ * \return `False` if env_name is invalid or NULL
+ * \return `False` on failure
+ */
```

Comment:
I think its important to point out the thread-safetyness of this function, both with respect to itself and with respect to concurrent changes when get_env when running.  i.e. the ongoing saga of https://github.com/ros2/rcutils/issues/242 .

Reply:
This is really great point. Both MSDN and setenv(3) call out the lack of thread safety. Neither differentiate between other instances of the same call and calls to a reader function, however. We might be able to get a firm answer for Linux by looking at the source, but unless someone finds something documented for Windows, it's probably safest to guard against current set operations.

From what I'm reading in #242, a resolution would mean guards in both get and set functions. If it's alright with you, I'd like to document the lack of thread safety in this PR, and let #242 address the thread safety issues between get and set operations. From the sounds of it, adding this function is the first step in that process anyway, and I believe that the signature of the function I'm adding here is adequate for implementing the new lock (unlike with `rcutils_get_env`).

Reply:
> If it's alright with you, I'd like to document the lack of thread safety in this PR, and let #242 address the thread safety issues between get and set operations.

Yeah, that's exactly what I'm suggesting; just make sure to document it here.  We'll address the larger issues as part of #242

Reply:
Done in 4ba9470